src/core/Events.js

```diff
@@ -90,95 +90,97 @@ L.Evented = L.Class.extend({
 
 	// attach listener (without syntactic sugar now)
 	_on: function (type, fn, context) {
+		this._events = this._events || {};
+
+		/* get/init listeners for type */
+		var typeListeners = this._events[type];

```

Comment:
I splitted this up after doing benchmarking, where more lookups are done when using the concise syntax. It's a micro-optimisation though, might not be true for modern browsers anymore.